src/transport/SecureSessionTable.h

```diff
@@ -69,6 +67,50 @@ class SecureSessionTable
         return result != nullptr ? MakeOptional<SessionHandle>(*result) : Optional<SessionHandle>::Missing();
     }
 
+    /**
+     * Allocates a new secure session out of the internal resource pool with the
+     * specified session ID.  The returned secure session will not become active
+     * until the call to SecureSession::Activate.
+     *
+     * @returns allocated session on success, else failure
+     */
+    CHECK_RETURN_VALUE
+    Optional<SessionHandle> CreateNewSecureSession(uint16_t localSessionId)
```

Comment:
Is this generally meant for tests?  If so, worth documenting that.

Reply:
Yes it is.  Thanks for the careful read.  I will add comments.

Reply:
Done.